interview-questions.md

```diff
@@ -8,62 +8,63 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Researched answer:
+  Researched answer: After research, function parameters are listed in the function definition. Function arguments are the values passed to the function.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: From memory, .map() takes in a value, index and the array. Of the three, only value is required. It is the same with other methods like filter.
 
-  Researched answer:
+  Researched answer: After researching to verify my answer, I verified that my answer was correct. I pulled the syntax from w3 schools to paste here. - array.map(function(currentValue, index, arr), thisValue) -
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: From memory, map loops over and "does something" to each element it touches along with returning the original array. Filter makes a "decision" on each element it touches and returns a portion or subset of an array.
 
-  Researched answer:
+  Researched answer: After researching this answer, the answers found were just like mine with a slightly different wording.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: From memory, a function is a rusable code machine that a developer makes to complete a given task or tasks. Methods are also built by developers but they are "built in", common use functions that devlopers have decided are used so much that essentially "javascript function presets" were made and called methods. Why capitalize words over and over when you could simply create a built in method to achieve the same thing with a single method.
 
-  Researched answer:
+  Researched answer: After researching this answer, I was on target with my answer. I have nothing more that add here.
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: From memory, destructuring is the pulling of specific data from things like arrays and objects and can map them into a new variable for use somewhere.
 
-  Researched answer:
+  Researched answer: After researching I additionally learned destructuring is a "javascript expression" and is used to extract multiple properties from an array​ at a time.
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: From memory, hoisting is what javascript does by default. It basically means that a variable can be used before it is declared. So you can declare it after it was used.
 
-  Researched answer:
+  Researched answer: After researching this a bit, I learned the interpreter seems to move the declaration of functions, variables or classes to the top of their scope, prior to execution of the code. It is this moving to the top of their scope, as I understand it, that makes it so that you can safely call on a variable before it is declared.
```

Comment:
For sure, although it is best practice to declare your variables before they are called.  Many other languages do not support hoisting and can cause bugs.